assets/activation-scripts/activate

```diff
@@ -180,7 +180,8 @@ fi
 # The rust CLI contains sophisticated logic to set $FLOX_SHELL based on the
 # process listening on STDOUT, but that won't happen when activating from
 # the top-level activation script, so fall back to $SHELL as a default.
-declare -r _flox_shell="${FLOX_SHELL:-$SHELL}"
+# Respect FLOX_SHELL_NOCLOBBER if set.
+declare -r _flox_shell="${FLOX_SHELL_NOCLOBBER:-${FLOX_SHELL:-$SHELL}}"
```

Comment:
It took me a few reads of [this thread](https://flox-dev.slack.com/archives/C05P6A5J6U8/p1736352763241429) to understand the rationale for this.

I conceptually understand the risk of "letting" users modify `SHELL` on-the-fly but I'm still not quite clear why we can't use it if our only purpose is for integration tests like https://github.com/flox/flox/pull/2576. Especially when we already have to isolate it for our integration tests per d570d6fc3e11809a0200a6976cfd1f7baba3e6e7.

I know it's only a small change, but it's yet another knob that we have to reason about. At a minimum I'd prefer "force" rather than "noclobber" to convey the purpose.

Reply:
I mentioned to Matthew already (maybe in a slack thread?) but FLOX_SHELL is already a force; the function of this new variable is to implement this force in a persistent way, so FLOX_SHELL_PERSIST would be another possibility. The "noclobber" word dates back to the earliest shells and refers to clobbering the contents of files, so agree that's not exactly the word here.

So do you prefer FLOX_SHELL_PERSIST?

Reply:
I suggested "force" because "persist" hints to me that we're doing extra work to further set the value rather than just not unsetting it. I'd take either over "noclobber" though 👍🏻 

Should we be converting all integration tests to use this new setting instead of `FLOX_SHELL` to avoid situations like https://github.com/flox/flox/pull/2635#discussion_r1927258356?

Reply:
Great, will use _FLOX_SHELL_PERSIST

Will leave conversion of other integration tests as a followup :+1: